metricsaggregator/src/main/java/io/dockstore/metricsaggregator/helper/RunExecutionAthenaAggregator.java

```diff
@@ -0,0 +1,211 @@
+
+    public void addGroupFields(Set<Field<?>> newGroupFields) {
+        this.groupFields.addAll(newGroupFields);
+    }
+
+    /**
+     * Create the runexecutions query string using the SELECT and GROUP BY fields.
+     *
+     * @return
+     */
+    @Override
+    protected String createQuery(AthenaTablePartition partition) {
+        // Sub-query that flattens the runexecutions array for the partition
+        final CommonTableExpression<Record7<String, String, String, Integer, Double, Integer, String>> executionsTable = name("executions")
+                .fields(PLATFORM_FIELD.getName(), DATE_EXECUTED_FIELD.getName(), EXECUTION_STATUS_FIELD.getName(), EXECUTION_TIME_SECONDS_FIELD.getName(),
+                        MEMORY_REQUIREMENTS_GB_FIELD.getName(), CPU_REQUIREMENTS_FIELD.getName(), COST_FIELD.getName())
+                .as(select(PLATFORM_FIELD,
+                        field("unnested." + DATE_EXECUTED_FIELD, String.class),
+                        field("unnested." + EXECUTION_STATUS_FIELD, String.class),
+                        field("unnested." + EXECUTION_TIME_SECONDS_FIELD, Integer.class),
+                        field("unnested." + MEMORY_REQUIREMENTS_GB_FIELD, Double.class),
+                        field("unnested." + CPU_REQUIREMENTS_FIELD, Integer.class),
+                        field("unnested." + COST_FIELD, String.class))
+                        .from(table(tableName), unnest(field("runexecutions", String[].class)).as("t", "unnested"))
+                        .where(ENTITY_FIELD.eq(inline(partition.entity()))
+                                .and(REGISTRY_FIELD.eq(inline(partition.registry())))
+                                .and(ORG_FIELD.eq(inline(partition.org())))
+                                .and(NAME_FIELD.eq(inline(partition.name())))
+                                .and(VERSION_FIELD.eq(inline(partition.version())))));
+        // This query created a workflow execution from each array of tasks. This will be unioned with the actual workflow executions submitted
+        final CommonTableExpression<Record7<String, String, String, Integer, Double, Integer, String>> runExecutionsFromTasksTable = name("runexecutionfromtasks")
+                .fields(PLATFORM_FIELD.getName(), DATE_EXECUTED_FIELD.getName(), EXECUTION_STATUS_FIELD.getName(), EXECUTION_TIME_SECONDS_FIELD.getName(),
+                        MEMORY_REQUIREMENTS_GB_FIELD.getName(), CPU_REQUIREMENTS_FIELD.getName(), COST_FIELD.getName())
+                .as(select(PLATFORM_FIELD,
+                        field("array_min(transform(taskexecutions, t -> t.dateexecuted))", String.class).as(DATE_EXECUTED_FIELD),
+                        // If all tasks are successful, set the workflow execution status as successful. Otherwise, assume failed
+                        field("case when all_match(transform(taskexecutions, t -> t.executionstatus), t -> t = 'SUCCESSFUL') then 'SUCCESSFUL' else 'FAILED' end", String.class).as(EXECUTION_STATUS_FIELD),
```

Comment:
This is where the task execution status aggregation differs from the Java version:
https://github.com/dockstore/dockstore-support/blob/bf345fd1aeb613e913e98165fe8c4c969261c05b/metricsaggregator/src/main/java/io/dockstore/metricsaggregator/helper/ExecutionStatusAggregator.java#L49-L53

There's no clean way to replicate it in SQL so if all task statuses are not successful, I set it to a generic fail.